server/src/DBTablesMonitoring.cc

```diff
@@ -1621,27 +1630,32 @@ bool DBTablesMonitoring::getTriggerInfo(TriggerInfo &triggerInfo,
 void DBTablesMonitoring::getTriggerInfoList(TriggerInfoList &triggerInfoList,
 					 const TriggersQueryOption &option)
 {
-	// build a condition
-	string condition = option.getCondition();
-	if (DBHatohol::isAlwaysFalseCondition(condition))
-		return;
+	DBClientJoinBuilder builder(tableProfileTriggers, &option);
+	builder.add(IDX_TRIGGERS_SERVER_ID);
+	builder.add(IDX_TRIGGERS_ID);
+	builder.add(IDX_TRIGGERS_STATUS);
+	builder.add(IDX_TRIGGERS_SEVERITY);
+	builder.add(IDX_TRIGGERS_LAST_CHANGE_TIME_SEC);
+	builder.add(IDX_TRIGGERS_LAST_CHANGE_TIME_NS);
+	builder.add(IDX_TRIGGERS_HOST_ID);
+	builder.add(IDX_TRIGGERS_HOSTNAME);
+	builder.add(IDX_TRIGGERS_BRIEF);
 
-	DBAgent::SelectExArg arg(tableProfileTriggers);
-	arg.tableField = option.getFromClause();
+	builder.addTable(
+	 tableProfileHosts, DBClientJoinBuilder::LEFT_JOIN,
+	 tableProfileTriggers,IDX_TRIGGERS_SERVER_ID,IDX_HOSTS_SERVER_ID,
+	 tableProfileTriggers,IDX_TRIGGERS_HOST_ID,IDX_HOSTS_HOST_ID);
+	
+	DBAgent::SelectExArg &arg = builder.build();
 	arg.useDistinct = option.isHostgroupUsed();
 	arg.useFullName = option.isHostgroupUsed();
-	arg.add(IDX_TRIGGERS_SERVER_ID);
-	arg.add(IDX_TRIGGERS_ID);
-	arg.add(IDX_TRIGGERS_STATUS);
-	arg.add(IDX_TRIGGERS_SEVERITY);
-	arg.add(IDX_TRIGGERS_LAST_CHANGE_TIME_SEC);
-	arg.add(IDX_TRIGGERS_LAST_CHANGE_TIME_NS);
-	arg.add(IDX_TRIGGERS_HOST_ID);
-	arg.add(IDX_TRIGGERS_HOSTNAME);
-	arg.add(IDX_TRIGGERS_BRIEF);
 
 	// condition
-	arg.condition = condition;
+	string optCond;
+	arg.condition.swap(optCond);
```

Comment:
Why do you use swap() ?
Can we pass arg.condition to isAlwaysFalseCondition() ?